### DISCUSSION TITLE
Block sharing

### DISCUSSION THREAD

**Discussion by simjnd** [warpdotdev/Warp#637] [Product ideas]
**Is your feature request related to a problem? Please describe.**
Sharing blocks is cool but limited and very static, asciinema is awesome and there is a real opportunity to integrate something similar in Warp and make it awesome.

**Describe the solution you'd like**
A better integrated version of https://asciinema.org/, that supports blocks. You could start recording a terminal session that will record the IO while preserving blocks. You can then share it just like you can currently share a block, the webpage would then play the session gradually revealing input and outputs of successive blocks. That could be very powerful when paired up with a button on the webpage to directly execute the session in Warp (mentioned in #54): you could not only easily share but also execute entire scripts. That can be great when following tutorials / walkthroughs where you have to repeat commands.

**Additional context**
N/A

**Comment by charlespierce** [comment]
@simjnd Thanks for the suggestion! I think this dovetails nicely with our planned work on shared sessions, we'll definitely need to think about these use-cases as well!

**Comment by joncalhoun** [comment]
The education space could really benefit from some similar thought.

For instance, it would be nice if 1+ blocks were exportable/embeddable in some way that allowed developers to intermix it with a written tutorial. Then as they read the tutorial, clicking any embedded block could open up the warp app and insert the text making it easy for users to tweak and hit enter.

It could also be a nice marketing opportunity if a bunch of write-ups start using embeds that advertise Warp.

**Comment by elviskahoro** [comment]
from fibric via Discord:
https://discord.com/channels/851854972600451112/906269689116770304/929684503935483955
> I would like to have a screenshot option for sharing commands and/or outputs to social media. I know macOS has a screenshot feature, but it captures “only” the window, not the scrollable content. Since Warp has gradient background design, stitching screenshots looks poor and amateur like.

> The social media world needs to see the beautiful design of Warp too

**Reply by nerdalytics** [reply]
asciicinema, shared blocks, etc. all go well with certain media and virtually infinite block size. Sharing screens works best in social media and handbooks, or when data privacy disallows remote sharing.

Taking a screenshot with macOS built-in functions captures the entire window or manually drawn capture frame. Larger console output requites screenshot stitching, but that only looks okay when not using gradient-based themes.

Since I'm a Fish shell user, I'm facing scrolled output a lot.

**Comment by elviskahoro** [comment]
Block sharing - links, expiration, and profile association

From @elviskahoro via #181

### Describe the solution you'd like?

```
When sharing a block through links ability to configure:
When it expires (various lengths + no expiration)
If choose to associate link with web profile, you'd be able to manage and view blocks you've shared in through some history dashboard (quota limits depending on pay-grade, free vs pro vs enterprise)
Anonymous block sharing links can have an auto-expiration up to a certain amount (defaults to 24 hours, but up to 72 hours or something like this)
```

### Is your feature request related to a problem? Please describe.
```
UIUX
```

### Additional context
```
Sourced from Discord: (General Channel - 08/12/2021)
Note: Extended core concept with some of personal ideas.
Original: "I think it could be cool if saving those commands to a link would show up on our profile, so we can manage and delete them. Perhaps, they auto expire after a week, but paying you can customize the delay?

I think stuff like that, and everything you mentioned i personally would be happy to pay for (if the price is right)."
```

**Comment by elviskahoro** [comment]
From dmiller via Discord:
https://discord.com/channels/851854972600451112/852249738002956311/935636056206897254
> I wish I could link to specific lines in a shared block 🙂
